src/libraries/Microsoft.Extensions.Hosting/src/HostOptions.cs

```diff
@@ -10,6 +11,16 @@ namespace Microsoft.Extensions.Hosting
     /// </summary>
     public class HostOptions
     {
+        public HostOptions()
+        {
+            var timeoutSeconds = Environment.GetEnvironmentVariable("DOTNET_SHUTDOWNTIMEOUTSECONDS");
+            if (!string.IsNullOrWhiteSpace(timeoutSeconds)
+                && int.TryParse(timeoutSeconds, NumberStyles.None, CultureInfo.InvariantCulture, out var seconds))
+            {
+                ShutdownTimeout = TimeSpan.FromSeconds(seconds);
```

Comment:
Test below was manually setting `ShutdownTimeout`

https://github.com/dotnet/runtime/blob/ae908fec2549ae2e901124794faf2f2233461ec8/src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/Internal/HostTests.cs#L599-L629

Let's get a test covering the case you are adding here.

Reply:
But how I supposed to test my case if I get value directly from Env, not from `IConfiguration`? I mean I see no way to mock Environment

Reply:
you're right. let me check.

@HaoK do you have a recommendation for this?

Reply:
I have to first investigate how this is handled this for "ASPNET_SHUTDOWNTIMEOUTSECONDS".

I think the approach needs to change a bit not just the test.

Reply:
"ASPNET_SHUTDOWNTIMEOUTSECONDS" does not work for Generic Host, created separated issue #44101